geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/api/MembershipListener.java

```diff
@@ -71,4 +71,7 @@ void memberSuspect(ID suspect, ID whoSuspected,
    */
   void saveConfig();
 
+  default Throwable setShutdownCause(Throwable t) {
```

Comment:
I can add javadoc for this method. 
Maybe the method name can be changed to setPossibleShutdownCauseToForceDisconnectException(shutdownCause)

The method signature is from the refactoring above. Since the original code expects to convert throwable into ForceDisconnectionException, thus this setShutdownCause() should return a Throwable. 

From listener point of view, we don't need to return anything. We did this way is only to avoid duplicated code. 

Of cause, I can complete revoked the code changes in ClusterDistributionManager.java, then I can change to void setShutdownCause(Throwable t) here, because in GMSMembership.java, we don't need setShutdownCause() to return anything.  

However, in our code, there will be 2 places with very similar code, one in ClusterDistributionManager, one in GMSMembership.  That's what we want to avoid. 

Conceptually, the whole fix's idea is to move setShutdownCause() a little earlier in force disconnect case. So it makes sense to do the refactoring in ClusterDistributionManager.DMListener into a method, then let listener to make use of this method.